src/layout.c

```diff
@@ -686,6 +687,81 @@ int igraph_layout_lgl(const igraph_t *graph, igraph_matrix_t *res,
 
 }
 
+int igraph_i_layout_reingold_tilford_unreachable(
+    const igraph_t *graph,
+    igraph_neimode_t mode,
+    long int real_root,
+    long int no_of_nodes,
+    igraph_vector_t *pnewedges);
+int igraph_i_layout_reingold_tilford_unreachable(
+    const igraph_t *graph,
+    igraph_neimode_t mode,
+    long int real_root,
+    long int no_of_nodes,
+    igraph_vector_t *pnewedges) {
+
+  long int no_of_newedges;
+  igraph_vector_t visited;
+  long int i, j, n;
+  igraph_dqueue_t q=IGRAPH_DQUEUE_NULL;
+  igraph_adjlist_t allneis;
+  igraph_vector_int_t *neis;
+
+  igraph_vector_resize(pnewedges, 0);
+
+  /* traverse from real_root and see what nodes you cannot reach */
+  no_of_newedges = 0;
+  IGRAPH_VECTOR_INIT_FINALLY(&visited, no_of_nodes);
+  IGRAPH_DQUEUE_INIT_FINALLY(&q, 100);
+
+  IGRAPH_CHECK(igraph_adjlist_init(graph, &allneis, mode));
+  IGRAPH_FINALLY(igraph_adjlist_destroy, &allneis);
+
+  /* start from real_root and go BFS */
+  IGRAPH_CHECK(igraph_dqueue_push(&q, real_root));
+  while (!igraph_dqueue_empty(&q)) {
+    long int actnode=(long int) igraph_dqueue_pop(&q);
+    neis=igraph_adjlist_get(&allneis, actnode);
+    n = igraph_vector_int_size(neis);
+    VECTOR(visited)[actnode] = 1;
+    for (j=0; j<n; j++) {
+      long int neighbor=(long int) VECTOR(*neis)[j];
+      if (!(long int)VECTOR(visited)[neighbor]) {
+        IGRAPH_CHECK(igraph_dqueue_push(&q, neighbor));
+      }
+    }
+  }
+
+  for (j=0; j<no_of_nodes; j++) {
+    no_of_newedges += 1 - VECTOR(visited)[j];
+  }
+
+  /* if any nodes are unreachable, add edges between them and real_root */
+  if (no_of_newedges != 0) {
+  
+    igraph_vector_resize(pnewedges, no_of_newedges * 2);
+    j = 0;
+    for (i=0; i<no_of_nodes; i++) {
+      if (!VECTOR(visited)[i]) {
+        if (mode!=IGRAPH_IN) {
+          VECTOR(*pnewedges)[2*j] = real_root;
+          VECTOR(*pnewedges)[2*j+1] = i;
+        } else {
+          VECTOR(*pnewedges)[2*j] = i;
+          VECTOR(*pnewedges)[2*j+1] = real_root;
+        }
+        j++;
+      }
+    }
+  }
+
+  igraph_dqueue_destroy(&q);
+  igraph_adjlist_destroy(&allneis);
+  igraph_vector_destroy(&visited);
+  IGRAPH_FINALLY_CLEAN(3);
+}
```

Comment:
There is now no proper return value, you have to return `IGRAPH_SUCCESS`. Doing so solves the out of memory error I believe (this is just a "random" error, depending on what value was returned).